chunk/schema.go

```diff
@@ -394,41 +224,45 @@ func (originalEntries) GetWriteEntries(_, _ uint32, tableName, hashKey string, l
 		}
 		result = append(result, IndexEntry{
 			TableName:  tableName,
-			HashValue:  hashKey,
+			HashValue:  bucketHashKey + ":" + string(metricName),
 			RangeValue: buildRangeKey([]byte(key), []byte(value), chunkIDBytes),
 		})
 	}
 	return result, nil
 }
 
-func (originalEntries) GetReadMetricEntries(_, _ uint32, tableName, hashKey string) ([]IndexEntry, error) {
-	return []IndexEntry{
+func (originalEntries) GetReadQueries(_, _ uint32, _, _ string) ([]IndexQuery, error) {
+	return nil, fmt.Errorf("originalEntries does not support GetReadQueries")
```

Comment:
Its probably time to make an ErrNoMetricName constant at the top of the file and return that.  Plus, lets make the error message something a user can understand "metric name required for pre-v7 schemas" or something.